### DISCUSSION TITLE
Need Help: E2E Testing Hindered by Google Authentication Login

### DISCUSSION THREAD

**Discussion by shirotamaki** [nextauthjs/next-auth#9773] [Help]
### Question 💬

Hello.
I need your help.

#### Situation
- I have created a web app using Rails 7 and Next.js 13.
- The app includes a login functionality using Google Authentication with next-auth.

```
"next": "^13.4.6",
"next-auth": "^4.22.1",
"react": "18.2.0",
"typescript": "5.0.2",

"cypress": "^12.17.1",
```

```
ruby '3.2.1'
gem 'rails', '~> 7.0.4', '>= 7.0.4.2'
```

#### What I want to do
- I am looking to conduct E2E testing of the web app using Cypress.
- There is no need to test the Google Authentication login function itself.
- However, I want to keep Google Authentication login operational as it enables E2E testing of the web app operations. As of now, since I am unable to log in, I am unable to conduct E2E testing of the app.

#### Problem
- If the Google Authentication login cannot be executed, users cannot access the web app.
- The web app services can only be operated by users who have logged in via Google Authentication.

#### What I tried
- I tried to implement it with reference to the following documents, but it was not successful.
  - https://docs.cypress.io/guides/end-to-end-testing/google-authentication#__docusaurus_skipToContent_fallback
  - https://next-auth.js.org/tutorials/testing-with-cypress
  - https://github.com/nextauthjs/next-auth/discussions/2053#discussioncomment-1191016

Please advise me on the best way to achieve my objectives.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by stale** [comment]
It looks like this issue did not receive any activity for 60 days. It will be closed in 7 days if no further activity occurs. If you think your issue is still relevant, commenting will keep it open. Thanks!

**Comment by MichaelRai00** [comment]
I'm facing the same issue
Right now the application I'm working on uses Google Authentication SSO 
and because of that we have to build a backdoor API login to log into the application - but the constrain is that we are using the same tenant - we can't create and run tests on new tenants
